test/framework/docker_logcollector.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kind/pkg/errors"
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// DockerLogCollector collect logs from a CAPD workload cluster.
+type DockerLogCollector struct{}
+
+func (k DockerLogCollector) CollectMachineLog(ctx context.Context, managementClusterClient client.Client, m *clusterv1.Machine, outputPath string) error {
+	containerName := fmt.Sprintf("%s-%s", m.Spec.ClusterName, m.Name)
+	execToPathFn := func(outputFileName, command string, args ...string) func() error {
+		return func() error {
+			f, err := fileOnHost(filepath.Join(outputPath, outputFileName))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			return execOnContainer(containerName, f, command, args...)
+		}
+	}
+	return errors.AggregateConcurrent([]func() error{
+		// record info about the node container
```

Comment:
What do you mean here?

Reply:
What I was thinking about is that in CAPD the node == a docker container, but I can remove this line

Reply:
Got it, yeah seems it's clear what we're collecting, up to you